specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-07-01-preview/dataConnectorDefinitions.json

```diff
@@ -0,0 +1,758 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ConnectorDefinitionBaseArmCollectionWrapper"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/dataConnectorDefinitions/{dataConnectorDefinitionName}": {
+      "get": {
+        "x-ms-examples": {
+          "Get customize data connector definition": {
+            "$ref": "./examples/dataConnectorDefinitions/GetCustomizableDataConnectoeDefinitionById.json"
+          }
+        },
+        "tags": [
+          "ConnectorDefinitions"
+        ],
+        "description": "Gets a data connector definition.",
+        "operationId": "DataConnectorDefinitions_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "in": "path",
+            "name": "dataConnectorDefinitionName",
+            "description": "The data connector definition name.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z0-9-]*$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ConnectorDefinitionBaseArmObjectWrapper"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
```

Comment:
There are definitions of error responses and details in commontypes types.json that you can use instead of declaring your own.
See e.g. `ErrorResponse` and `ErrorDetail` in
https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v5/types.json

Reply:
Please note that the Schema ref is to a common Error type. 
i just add here the description, before the reference to the common type. we have the same in all our json files. 
Anything i need to change in description or the way i reference the common type?